src/navsat_transform.cpp

```diff
@@ -686,20 +687,20 @@ bool NavSatTransform::prepareFilteredGps(
   return new_data;
 }
 
-bool NavSatTransform::prepareGpsOdometry(nav_msgs::msg::Odometry & gps_odom)
+bool NavSatTransform::prepareGpsOdometry(nav_msgs::msg::Odometry * gps_odom)
 {
   bool new_data = false;
 
   if (transform_good_ && gps_updated_ && odom_updated_) {
-    gps_odom = utmToMap(latest_utm_pose_);
+    *gps_odom = std::move(utmToMap(latest_utm_pose_));
```

Comment:
Please make sure this has the intended consequences. Its very hard to tell if this will work without extensive coverage of unit testing on this front. Can you make sure that this message is being properly populated?

Reply:
I did pass all the unit tests but I do understand your concerns. Just in this one case is it better to pass unique_ptr to this function? Because i have learnt returning smart pointers is not a good practice.

Reply:
I'm wondering why not `*gps_odom = utmToMap(latest_utm_pose_);`? The std::move I think removes the memory from latest_utm_pose_ and navsat transform has alot of moving parts happening asynchronously. I'm not 100% rock-solid confident that that won't have other effects. 

I guess can you talk about why you did that here and maybe that will help me understand

Reply:
I think move semantics dictate that objects created in methods automatically get moved when returned. `*gps_odom = utmToMap(latest_utm_pose_);` should work fine, I think.

Reply:
> I think move semantics dictate that objects created in methods automatically get moved when returned.

Correct me if I am wrong. `utmToMap(latest_utm_pose_)` is an rvalue and I wanted to move the temporary data (scope limited to this function) created to the pointer.

>  Navsat transform has alot of moving parts happening asynchronously.

True. Not a 100% sure but this should work fine. I'll change this to 
`*gps_odom = utmToMap(latest_utm_pose_)` just to be one the safe side.